metronome/config/src/io/iohk/metronome/config/ConfigDecoders.scala

```diff
@@ -35,6 +35,40 @@ object ConfigDecoders {
       tryParse(_, _ getBoolean _)
     }
 
+  /**
+    * Custom sequence decoder with failover (Map[Int, A]) handles system property overrides
+    *
+    * It turned out that:
+    * ```
+    *   "field": ["valueA", "valueB", "valueC"]
+    *
+    *   // when overridden as:
+    *   ./run -Dfield.0="valueX"
+    *
+    *   // turns into:
+    *   "field": { "0": "valueX" }
+    * ```
+    * Failing the cursor expecting Json array.
+    * However this situation can be handled by reading an Object instead.
+    *
+    * @param aDecoder - decoder for each element
+    * @return Seq (Vector) of A's
+    */
+
+  implicit def seqDecoder[A](implicit aDecoder: Decoder[A]): Decoder[Seq[A]] = {
+    vectorDecoder.map(_.toSeq) // returns vector itself, actually
+  }
+
+  implicit def listDecoder[A](implicit aDecoder: Decoder[A]): Decoder[List[A]] = {
+    vectorDecoder.map(_.toList)
+  }
+
+  implicit def vectorDecoder[A](implicit aDecoder: Decoder[A]): Decoder[Vector[A]] = {
+    Decoder.decodeVector[A] or Decoder[Map[Int, A]].map { srcMap =>
+      srcMap.toVector.sortBy(_._1).map(_._2)
```

Comment:
So we discussed on Slack that this could check that indices actually go from `0..n` in a contiguous fashion, so for example the user does not live under the impression that they have overridden just item `arr.2` when they have in fact reduced their array to `[arr.2]` and discarded all other default items. 

You seemed to agree there.

Reply:
Yep, now I cleaned it up, made some tests that detect such override mistakes. Thanks!